Source/SwiftLintFramework/Rules/LineLengthRule.swift

```diff
@@ -87,4 +91,25 @@ public struct LineLengthRule: ConfigurationProviderRule, SourceKitFreeRule {
         return modifiedString
     }
 
+    private func stripUrls(fromSourceString sourceString: String) -> String {
+        let range = NSRange(location: 0, length: sourceString.bridge().length)
+        // Workaround for Linux until NSDataDetector is available
+        #if os(Linux)
+            // Regex pattern from http://daringfireball.net/2010/07/improved_regex_for_matching_urls
+            let pattern = "(?i)\\b((?:[a-z][\\w-]+:(?:/{1,3}|[a-z0-9%])|www\\d{0,3}[.]|[a-z0-9.\\-]+[.][a-z]{2,4}/)" +
+                "(?:[^\\s()<>]+|\\(([^\\s()<>]+|(\\([^\\s()<>]+\\)))*\\))+(?:\\(([^\\s()<>]+|(\\([^\\s()<>]+\\)))*" +
+                "\\)|[^\\s`!()\\[\\]{};:'\".,<>?«»“”‘’]))"
+            guard let regex = try? NSRegularExpression(pattern: pattern, options: []) else {
```

Comment:
you could use `regex(pattern)` here so it can be cached (and you know it it's a valid regex)